File: src/main/java/mods/cybercat/gigeresque/client/entity/render/RunnerAlienEntityRenderer.java
package mods.cybercat.gigeresque.client.entity.render;

import com.mojang.blaze3d.vertex.PoseStack;
import mod.azure.azurelib.common.api.client.renderer.GeoEntityRenderer;
import mods.cybercat.gigeresque.client.entity.model.RunnerAlienEntityModel;
import mods.cybercat.gigeresque.client.entity.render.feature.RunnerAlienFeatureRenderer;
import mods.cybercat.gigeresque.common.entity.impl.runner.RunnerAlienEntity;
import net.minecraft.client.renderer.MultiBufferSource;
import net.minecraft.client.renderer.entity.EntityRendererProvider;
import org.jetbrains.annotations.NotNull;

public class RunnerAlienEntityRenderer extends GeoEntityRenderer<RunnerAlienEntity> {
    public RunnerAlienEntityRenderer(EntityRendererProvider.Context context) {
        super(context, new RunnerAlienEntityModel());
        this.shadowRadius = 0.5f;
        this.addRenderLayer(new RunnerAlienFeatureRenderer(this));
    }

    @Override
    public void render(RunnerAlienEntity entity, float entityYaw, float partialTicks, PoseStack stack, @NotNull MultiBufferSource bufferIn, int packedLightIn) {
        float scaleFactor = 0.5f + ((entity.getGrowth() / entity.getMaxGrowth()) / 5f);
        stack.scale(scaleFactor, scaleFactor, scaleFactor);
        if (!entity.isVehicle()) {
            stack.translate(0, -0.16, 0);
        }
        super.render(entity, entityYaw, partialTicks, stack, bufferIn, packedLightIn);
    }

    @Override
    protected float getDeathMaxRotation(RunnerAlienEntity entityLivingBaseIn) {
        return 0;
    }

    @Override
    public float getMotionAnimThreshold(RunnerAlienEntity animatable) {
        return 0.005f;
    }
}

File: src/main/kotlin/com/bvanseg/gigeresque/client/entity/render/RunnerAlienEntityRenderer.kt
package com.bvanseg.gigeresque.client.entity.render

import com.bvanseg.gigeresque.client.entity.model.RunnerAlienEntityModel
import com.bvanseg.gigeresque.client.entity.render.feature.RunnerAlienFeatureRenderer
import com.bvanseg.gigeresque.client.extensions.scale
import com.bvanseg.gigeresque.common.entity.impl.RunnerAlienEntity
import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment
import net.minecraft.client.render.VertexConsumerProvider
import net.minecraft.client.render.entity.EntityRendererFactory
import net.minecraft.client.util.math.MatrixStack
import software.bernie.geckolib3.renderers.geo.GeoEntityRenderer

/**
 * @author Boston Vanseghi
 */
@Environment(EnvType.CLIENT)
class RunnerAlienEntityRenderer(context: EntityRendererFactory.Context) : GeoEntityRenderer<RunnerAlienEntity>(
    context,
    RunnerAlienEntityModel()
) {
    init {
        this.shadowRadius = 0.5f
        this.addLayer(RunnerAlienFeatureRenderer(this))
    }

    override fun render(
        entity: RunnerAlienEntity,
        entityYaw: Float,
        partialTicks: Float,
        stack: MatrixStack,
        bufferIn: VertexConsumerProvider,
        packedLightIn: Int
    ) {
        stack.scale(0.5f + ((entity.growth / entity.maxGrowth.toFloat()) / 5f))
        stack.translate(0.0, 0.1, 0.0)
        super.render(entity, entityYaw, partialTicks, stack, bufferIn, packedLightIn)
    }
}